src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs

```diff
@@ -0,0 +1,325 @@
+        /// <typeparam name="T">Type of the data we should retrieve</typeparam>
+        /// <returns>The API response in a GoFeatureFlagResponse object.</returns>
+        /// <exception cref="FlagNotFoundError">If the flag does not exists</exception>
+        /// <exception cref="GeneralError">If an unknown error happen</exception>
+        /// <exception cref="FlagDisabled">If the flag is disabled</exception>
+        private async Task<GoFeatureFlagResponse> CallApi<T>(string flagKey, T defaultValue,
+            EvaluationContext context = null)
+        {
+            var user = GoFeatureFlagUser.FromEvaluationContext(context);
+            var request = new GOFeatureFlagRequest<T>
+            {
+                User = user,
+                DefaultValue = defaultValue
+            };
+            var goffRequest = JsonSerializer.Serialize(request, _serializerOptions);
+
+            var response = await _httpClient.PostAsync("v1/feature/" + flagKey + "/eval",
+                new StringContent(goffRequest, Encoding.UTF8, ApplicationJson));
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
+                throw new FlagNotFoundError("Flag " + flagKey + " was not found in your configuration");
+
+            if (response.StatusCode >= HttpStatusCode.BadRequest)
+                throw new GeneralError("impossible to contact GO Feature Flag relay proxy instance");
+
+            var responseBody = await response.Content.ReadAsStringAsync();
+            var goffResp =
+                JsonSerializer.Deserialize<GoFeatureFlagResponse>(responseBody);
+
+            if (goffResp != null && Reason.Disabled.Equals(goffResp.reason))
+                throw new FlagDisabled();
+
+            if (ErrorType.FlagNotFound.ToString().Equals(goffResp.errorCode))
+                throw new FlagNotFoundError("Flag " + flagKey + " was not found in your configuration");
+
+            return goffResp;
+        }
+
+        /// <summary>
+        ///     convertValue is converting the object return by the proxy response in the right type.
+        /// </summary>
+        /// <param name="value">The value we have received</param>
+        /// <returns>A converted object</returns>
+        /// <exception cref="InvalidCastException">If we are not able to convert the data.</exception>
+        private Value ConvertValue(JsonElement value)
+        {
+            if (value.ValueKind == JsonValueKind.Null || value.ValueKind == JsonValueKind.Undefined) return null;
+
+            if (value.ValueKind == JsonValueKind.False || value.ValueKind == JsonValueKind.True)
+                return new Value(value.GetBoolean());
+
+            if (value.ValueKind == JsonValueKind.Number) return new Value(value.GetDouble());
+
+            if (value.ValueKind == JsonValueKind.Object)
+            {
+                var dict = new Dictionary<string, Value>();
+                var objEnumerator = value.EnumerateObject();
```

Comment:
`JsonElement.ObjectEnumerator` implements [IDisposable](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.objectenumerator.dispose?view=net-6.0#system-text-json-jsonelement-objectenumerator-dispose). I think you should be using it withing a `using` statement to make sure it's memory is released in a timely manner.

Reply:
That's the 1st time I write C# I did not even knew that `using` was a thing 🤦 .
Thanks for the help I will have a look at it.